tools/ncf_constraints.py

```diff
@@ -71,21 +71,33 @@ def allow_whitespace_string(parameter, allow):
     }
 }
 
+variable_constraints = {
+  "max_length" : constraints["max_length"]
+}
+
 default_constraint = {
     "allow_whitespace_string" : False
   , "allow_empty_string" : False
 }
 
-def check_parameter(parameter, parameter_constraints):
-  """Checks that a parameter is ok with the constraint passed as parameter"""
+def check_parameter(parameter_value, parameter_constraints):
+  """Checks that a parameter value is ok with the constraint of the parameter"""
 
   result = True
   errors = []
+  constraint_set = constraints
+
+  # Check that our value contains a variable or not
+  value_without_variables = re.sub(r'[\$@][\{\(][a-zA-Z0-9\[\]_.-]+[\}\)]', "", parameter_value)
```

Comment:
This won't match cfengine indexed arrays "${value[${index}]}"
To have a real variable parsing you need an expression parser, since we don't want an exact variable matching, but just know that a variable in present, you can just add ${( in the allowed character within the name